<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function will take two provided clusters and
assess whether they are distinguishable by a permutation test using random
forest classifier prediction accuracies."><title>Compare any two clusters using CHOIR's random forest classifier permutation testing approach — compareClusters • CHOIR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Compare any two clusters using CHOIR's random forest classifier permutation testing approach — compareClusters"><meta property="og:description" content="This function will take two provided clusters and
assess whether they are distinguishable by a permutation test using random
forest classifier prediction accuracies."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CHOIR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/CHOIR.html">
    <span class="fa fa-graduation-cap"></span>
     
    Quick Tutorial
  </a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">
    <span class="fa fa-code"></span>
     
    Functions
  </a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/corceslab/CHOIR">
    <span class="fa fa-github"></span>
     
    CHOIR on GitHub
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Compare any two clusters using CHOIR's random forest classifier permutation testing approach</h1>
      
      <div class="d-none name"><code>compareClusters.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function will take two provided clusters and
assess whether they are distinguishable by a permutation test using random
forest classifier prediction accuracies.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">compareClusters</span><span class="op">(</span>
  object <span class="op">=</span> <span class="cn">NULL</span>,
  key <span class="op">=</span> <span class="st">"CHOIR"</span>,
  cluster1_cells <span class="op">=</span> <span class="cn">NULL</span>,
  cluster2_cells <span class="op">=</span> <span class="cn">NULL</span>,
  ident1 <span class="op">=</span> <span class="cn">NULL</span>,
  ident2 <span class="op">=</span> <span class="cn">NULL</span>,
  group_by <span class="op">=</span> <span class="cn">NULL</span>,
  alpha <span class="op">=</span> <span class="fl">0.05</span>,
  feature_set <span class="op">=</span> <span class="st">"var"</span>,
  var_features <span class="op">=</span> <span class="cn">NULL</span>,
  exclude_features <span class="op">=</span> <span class="cn">NULL</span>,
  n_iterations <span class="op">=</span> <span class="fl">100</span>,
  n_trees <span class="op">=</span> <span class="fl">50</span>,
  use_variance <span class="op">=</span> <span class="cn">TRUE</span>,
  min_accuracy <span class="op">=</span> <span class="fl">0.5</span>,
  min_connections <span class="op">=</span> <span class="fl">1</span>,
  max_repeat_errors <span class="op">=</span> <span class="fl">20</span>,
  collect_all_metrics <span class="op">=</span> <span class="cn">FALSE</span>,
  sample_max <span class="op">=</span> <span class="cn">Inf</span>,
  downsampling_rate <span class="op">=</span> <span class="st">"auto"</span>,
  normalization_method <span class="op">=</span> <span class="st">"none"</span>,
  batch_labels <span class="op">=</span> <span class="cn">NULL</span>,
  use_assay <span class="op">=</span> <span class="cn">NULL</span>,
  use_slot <span class="op">=</span> <span class="cn">NULL</span>,
  ArchR_matrix <span class="op">=</span> <span class="cn">NULL</span>,
  atac <span class="op">=</span> <span class="cn">FALSE</span>,
  input_matrix <span class="op">=</span> <span class="cn">NULL</span>,
  nn_matrix <span class="op">=</span> <span class="cn">NULL</span>,
  n_cores <span class="op">=</span> <span class="cn">NULL</span>,
  random_seed <span class="op">=</span> <span class="fl">1</span>,
  verbose <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>object</dt>
<dd><p>An object of class 'Seurat', 'SingleCellExperiment', or 'ArchRProject'.
Not used if values are provided for parameters 'input_matrix' and
'nn_matrix'.</p></dd>


<dt>key</dt>
<dd><p>The name under which CHOIR-related data is retrieved from the
object. Defaults to 'CHOIR'. Not used if values are provided for parameters
'input_matrix' and 'nn_matrix'.</p></dd>


<dt>cluster1_cells</dt>
<dd><p>A character vector of cell names belonging to cluster 1.</p></dd>


<dt>cluster2_cells</dt>
<dd><p>A character vector of cell names belonging to cluster 2.</p></dd>


<dt>ident1</dt>
<dd><p>A string indicating the label for cluster 1.</p></dd>


<dt>ident2</dt>
<dd><p>A string indicating the label for cluster 2.</p></dd>


<dt>group_by</dt>
<dd><p>A string indicating the column of cluster labels that 'ident1'
and 'ident2' belong to.</p></dd>


<dt>alpha</dt>
<dd><p>A numerical value indicating the significance level used for
permutation test comparisons of cluster distinguishability. Defaults to 0.05.</p></dd>


<dt>feature_set</dt>
<dd><p>A string indicating whether to train random forest
classifiers on 'all' features or only variable ('var') features. Defaults to 'var'.</p></dd>


<dt>var_features</dt>
<dd><p>An optional character vector of variable features to be
used for random forest comparisons. Defaults to <code>NULL</code>.</p></dd>


<dt>exclude_features</dt>
<dd><p>A character vector indicating features that should be
excluded from input to the random forest classifier. Default = <code>NULL</code>
will not exclude any features.</p></dd>


<dt>n_iterations</dt>
<dd><p>A numeric value indicating the number of iterations run
for each permutation test comparison. Defaults to 100.</p></dd>


<dt>n_trees</dt>
<dd><p>A numeric value indicating the number of trees in each random
forest. Defaults to 50.</p></dd>


<dt>use_variance</dt>
<dd><p>A boolean value indicating whether to use the variance of
the random forest accuracy scores as part of the permutation test threshold.
Defaults to <code>TRUE</code>.</p></dd>


<dt>min_accuracy</dt>
<dd><p>A numeric value indicating the minimum accuracy required
of the random forest classifier, below which clusters will be automatically
merged. Defaults to 0.5.</p></dd>


<dt>min_connections</dt>
<dd><p>A numeric value indicating the minimum number of
nearest neighbors between two clusters for them to be considered 'adjacent'.
Non-adjacent clusters will not be merged. Defaults to 0.</p></dd>


<dt>max_repeat_errors</dt>
<dd><p>Used to account for situations in which random
forest classifier errors are concentrated among a few cells that are
repeatedly misassigned. Numeric value indicating the maximum number of such
'repeat errors' that will be taken into account. If set to 0, 'repeat errors'
will not be evaluated. Defaults to 20.</p></dd>


<dt>collect_all_metrics</dt>
<dd><p>A boolean value indicating whether to collect and save
additional metrics from the random forest classifiers, including feature
importances and tree depth. Defaults to <code>FALSE</code>.</p></dd>


<dt>sample_max</dt>
<dd><p>A numeric value indicating the maximum number of cells used
per cluster to train/test each random forest classifier. Default = <code>Inf</code>
does not cap the number of cells used.</p></dd>


<dt>downsampling_rate</dt>
<dd><p>A numeric value indicating the proportion of cells used
per cluster to train/test each random forest classifier. Default = "auto" sets
the downsampling rate according to the dataset size, for efficiency.</p></dd>


<dt>normalization_method</dt>
<dd><p>A character string or vector indicating which
normalization method to use. In general, input data should be supplied to
CHOIR after normalization, except in cases when the user wishes to use
<code><a href="https://satijalab.org/seurat/reference/SCTransform.html" class="external-link">Seurat::SCTransform()</a></code> normalization. Permitted values are 'none' or
'SCTransform'. Defaults to 'none'.</p></dd>


<dt>batch_labels</dt>
<dd><p>If applying batch correction, the name of the column
containing the batch labels. Defaults to <code>NULL</code>.</p></dd>


<dt>use_assay</dt>
<dd><p>For Seurat or SingleCellExperiment objects, a character
string or vector indicating the assay(s) to use in the provided object.
Default = <code>NULL</code> will choose the current active assay for Seurat objects
and the <code>log_counts</code> assay for SingleCellExperiment objects.</p></dd>


<dt>use_slot</dt>
<dd><p>For Seurat objects, a character string or vector indicating
the slot(s) (Seurat v4) or layer(s) (Seurat v5) to use in the provided object.
Default = <code>NULL</code> will choose a slot/layer based on the selected assay.
If a non-standard assay is provided, do not leave <code>use_slot</code> as <code>NULL</code>.</p></dd>


<dt>ArchR_matrix</dt>
<dd><p>For ArchR objects, a character string or vector
indicating which matri(ces) to use in the provided object. Default =
<code>NULL</code> will use the 'TileMatrix' for ATAC-seq data or the
'GeneExpressionMatrix' for RNA-seq data.</p></dd>


<dt>atac</dt>
<dd><p>A boolean value or vector indicating whether the provided data is
ATAC-seq data. Defaults to <code>FALSE</code>.</p></dd>


<dt>input_matrix</dt>
<dd><p>An optional matrix containing the feature x cell data on
which to train the random forest classifiers. Default = <code>NULL</code> will use
the feature x cell matri(ces) indicated by function <code><a href="buildTree.html">buildTree()</a></code>.</p></dd>


<dt>nn_matrix</dt>
<dd><p>An optional matrix containing the nearest neighbor adjacency
of the cells. Default = <code>NULL</code> will look for the adjacency matri(ces)
generated by function <code><a href="buildTree.html">buildTree()</a></code>.</p></dd>


<dt>n_cores</dt>
<dd><p>A numeric value indicating the number of cores to use for
parallelization. Default = <code>NULL</code> will use the number of available cores
minus 2.</p></dd>


<dt>random_seed</dt>
<dd><p>A numeric value indicating the random seed to be used.</p></dd>


<dt>verbose</dt>
<dd><p>A boolean value indicating whether to use verbose output
during the execution of this function. Can be set to <code>FALSE</code> for a
cleaner output.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>Returns a list containing the following elements:</p><dl><dt>comparison_result</dt>
<dd><p>A string, either "merge" or "split", indicating the result of the comparison.</p></dd>

<dt>comparison_records</dt>
<dd><p>A dataframe including the metrics recorded for the comparison</p></dd>

<dt>feature_importances</dt>
<dd><p>If 'collect_all_metrics' is true, a dataframe containing the feature importance scores for each gene in the comparison</p></dd>


</dl></div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Cathrine Petersen.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.4.</p>
</div>

    </footer></div>

  

  

  </body></html>

